hyperspy/io_plugins/hdf5.py

```diff
@@ -317,14 +318,14 @@ def parse_structure(key, group, value, _type, compression):
         except ValueError:
             tmp = np.array([[0]])
         if tmp.dtype is np.dtype('O') or tmp.ndim is not 1:
-            dict2hdfgroup(dict(zip(
-                [unicode(i) for i in xrange(len(value))], value)),
+            dict2hdfgroup(dict(list(zip(
+                [str(i) for i in range(len(value))], value))),
```

Comment:
I think we can skip the `list` wrap here? dict should be able to work with iterators, right?